libs/ledger/ledger/events.py

```diff
@@ -31,6 +31,14 @@ class EventSet(Ledger):
     detection_time: np.ndarray = parameter()
     shift: np.ndarray = parameter()
     Tb: float = metadata(default=0)
+    sorted_statistic: np.ndarray = parameter()
```

Comment:
If we're going to do the sorting here, it wouldn't make the underlying data in the file itself sorted and save us any time in the long term. 

Would it make more sense to sort at the end of the `Infer` task? Would require loading the files back in and sorting. Also, we have nice indexing functionality builtin already, so we can sort all the `fields` together. So the end of the Infer task would look something like:

```python
EventSet.aggregate(files, output)
events = EventSet.read(output)
args = np.argsort(events.detection_statistic)
events = events[args]
events.write(output)
```

Reply:
More longer term, a lot of the functionality we want  / have built comes for free with pandas DataFrames. 

The really useful functionality we have here is the out of memory merging via `aggregate`. Not sure how it would look, but we can probably build on top of pandas

Reply:
Agreed that we might as well just sort all the fields, but to me that functionality should occur within the object itself.  I forget that `aggregate` doesn't do something like `cls(**kwargs)` the way `read` does: the way it's set up now, I could just call `read` and `write` to sort things. I'll think some more.

Reply:
There are some cases where one might not want to wait for the sorting to occur (e.g plotting histograms) which makes me think it shouldn't occur in `__init__`. 

Maybe adding a method `.sort(attr: str)` that will sort the whole object by the passed attribute. 

We can also have a `kwarg` in `aggregate` which optionally returnsthe class itself by loading it back in

Reply:
Lol I just started writing up both of those as suggestions. I think no need to change `aggregate` if the sorting isn't happening on initialization anymore, though.

I'll add a method for sorting and something that checks if the object is sorted by a particular key.

Reply:
Yeah that's perfect. Should probably be part of the `Ledger` object. 

Then things like the `nb` method can check if it's sorted by the necessary key and use the optimized method, and otherwise do the sorting

Reply:
Think I've done what we talked about here, let me know what you think